Fix autogen.sh to exit with an error code if an error occurs.
https://github.com/huzheng001/stardict-3/pull/103
--- autogen.sh.orig
+++ autogen.sh
@@ -6,21 +6,21 @@ echo "Bootstrapping StarDict root..."
 # 	echo;
 # 	echo "You must have libtool installed to compile StarDict";
 # 	echo;
-# 	exit;
+# 	exit 1;
 # }
 
 (automake --version) < /dev/null > /dev/null 2>&1 || {
 	echo;
 	echo "You must have automake installed to compile StarDict";
 	echo;
-	exit;
+	exit 1;
 }
 
 (autoconf --version) < /dev/null > /dev/null 2>&1 || {
 	echo;
 	echo "You must have autoconf installed to compile StarDict";
 	echo;
-	exit;
+	exit 1;
 }
 
 echo "Generating configuration files for StarDict, please wait...."
@@ -40,13 +40,13 @@ aclocal -I m4 || exit;
 # echo "Running autoheader...."
 # autoheader || exit;
 echo "Running automake --add-missing --copy...."
-automake --add-missing --copy;
+automake --add-missing --copy || exit;
 echo "Running autoconf ...."
 autoconf || exit;
 echo "Running automake ...."
 automake || exit;
 cd "$topdir"
-${srcdir}/lib/autogen.sh
-${srcdir}/dict/autogen.sh
-${srcdir}/tools/autogen.sh
+${srcdir}/lib/autogen.sh || exit
+${srcdir}/dict/autogen.sh || exit
+${srcdir}/tools/autogen.sh || exit
 #"${srcdir}/configure" --prefix=/usr --sysconfdir=/etc --mandir=/usr/share/man --disable-deprecations --disable-gucharmap --disable-dictdotcn --disable-gnome-support --disable-scrollkeeper "$@"
--- lib/autogen.sh.orig
+++ lib/autogen.sh
@@ -6,21 +6,21 @@ echo "Boostrapping common lib..."
 	echo;
 	echo "You must have libtool installed to compile common lib";
 	echo;
-	exit;
+	exit 1;
 }
 
 (automake --version) < /dev/null > /dev/null 2>&1 || {
 	echo;
 	echo "You must have automake installed to compile common lib";
 	echo;
-	exit;
+	exit 1;
 }
 
 (autoconf --version) < /dev/null > /dev/null 2>&1 || {
 	echo;
 	echo "You must have autoconf installed to compile common lib";
 	echo;
-	exit;
+	exit 1;
 }
 
 echo "Generating configuration files for common lib, please wait...."
@@ -39,7 +39,7 @@ aclocal -I m4 || exit;
 echo "Running autoheader...."
 autoheader || exit;
 echo "Running automake --add-missing --copy...."
-automake --add-missing --copy;
+automake --add-missing --copy || exit;
 echo "Running autoconf ...."
 autoconf || exit;
 echo "Running automake ...."
--- tools/autogen.sh.orig
+++ tools/autogen.sh
@@ -6,21 +6,21 @@ echo "Boostrapping StarDict tools..."
 	echo;
 	echo "You must have libtool installed to compile Stardict";
 	echo;
-	exit;
+	exit 1;
 }
 
 (automake --version) < /dev/null > /dev/null 2>&1 || {
 	echo;
 	echo "You must have automake installed to compile Stardict";
 	echo;
-	exit;
+	exit 1;
 }
 
 (autoconf --version) < /dev/null > /dev/null 2>&1 || {
 	echo;
 	echo "You must have autoconf installed to compile Stardict";
 	echo;
-	exit;
+	exit 1;
 }
 
 echo "Generating configuration files for Stardict, please wait...."
@@ -39,7 +39,7 @@ aclocal -I m4 || exit;
 echo "Running autoheader...."
 autoheader || exit;
 echo "Running automake --add-missing --copy...."
-automake --add-missing --copy;
+automake --add-missing --copy || exit;
 echo "Running autoconf ...."
 autoconf || exit;
 echo "Running automake ...."
